### ISSUE TITLE
C-c a not working in emacx 28.0.50 on Mac OS 10.15.6

### ISSUE THREAD

**Issue by MooersLab** [freetonik/castlemacs#23]
This may be a Mac OS 10.15.6 issue.
I compiled emacs from source. I am using version 28.0.50.

I added to private.el

```
(global-set-key "\C-ca" 'org-agenda)
```
This is the only command in private.el.

This is happening with an otherwise clean and new install of castlemacs.

I tried opening an org file again.
The command C-c a still does not work.

*By the way, your README.org file is filled with useful information. It is very nicely done. Thank you very much for taking the trouble to put it together!!

**Comment by freetonik** [comment]
Hey,

what does it say if you hit `C-h k C-c a`?

**Comment by MooersLab** [comment]
`C-c a` is undefined.

**Comment by MooersLab** [comment]
As I edit an org document, I am getting in the messages buffer the following message, which may or may not be related.

```Error parsing SVG image ‘(image :type svg :file splash.svg :scale 1)’ [5 times]```

I am not seeing a splash.svg on start up but that does not bother me.

**Comment by freetonik** [comment]
It seems like it's related to the Emacs version. Your config works fine under macOS 10.14.16 + Emacs 26.3.

I try to support the latest stable release (currently 26.3). 27 should work fine, but with 28 there might be more problems.

I might need some time to investigate the compatibility with newer versions. But I'm afraid the easiest way to resolve this is to switch to a stable 26.3. Is there a reason you've decided to use v28?

**Comment by MooersLab** [comment]
I thought that some issues with Catalina might have been resolved in v28.
I was using v26.3.

I will switch back to v26.3 and test it with the castlemacs config. 
I have not tried that.

I have bounced off emacs several times in the past and then dug into vim.
After getting comfortable and productive in vim, I feel tough enough to try to emacs again.
Castlemacs may ease that transition.

**Comment by freetonik** [comment]
Ugh, I'm afraid to upgrade to Catalina still :(
But I guess I should, in order to properly support Castlemacs.

Let me know how it goes.

**Comment by MooersLab** [comment]
I am getting the same message, so the problem must be with a hidden configuration file outside of .init.el.

**Comment by freetonik** [comment]
Try putting `(global-set-key "\C-ca" 'org-agenda)` in a scratch buffer, placing the cursor at the end of the line and pressing `C-x C-e` (evaluate last sexp).

**Comment by MooersLab** [comment]
I get back `org-agenda`.

**Comment by freetonik** [comment]
Now try to invoke that key combination (or hit `C-h k C-c a` again).

**Comment by MooersLab** [comment]
Now I have the help page for org-agenda.

**Comment by freetonik** [comment]
Just to be clear, it says something like:

```
C-c a runs the command org-agenda (found in global-map), which is an
interactive autoloaded Lisp function in ‘org-agenda.el’.
....
```

?

**Comment by MooersLab** [comment]
Correct!

**Comment by freetonik** [comment]
Alright, so it seems like `private.el` is not loaded correctly. Are there any error messages in the minibuffer when Emacs loads?

**Comment by MooersLab** [comment]
Oops, I ignored this as I was focused on the second window with the org file in it.

Warning (initialization): An error occurred while loading ‘/Users/blaine/.emacs.d/init.el’:

error: Expected printf output from shell, but got: "[5 qINFO: deactivate-gfortran_osx-64.sh made the following environmental changes:
__RESULT/Users/blaine/.opam/default/bin:/opt/anaconda/condabin:/bin:/opt/local/bin:/opt/local/include:
/opt/local/Library/Frameworks/Python.framework/Versions/3.8/bin:/Users/blaine/Library/Python/3.8/bin:/Users/blaine/Library/Python/3.7/bin:/opt/local/Library/Frameworks/Python.framework/Versions/3.7/bin:/opt/local/Library/Frameworks/Python.framework/Versions/3.6/bin:/opt/local/Library/Frameworks/Python.framework/Versions/3.5/bin:/opt/local/libexec/gnubin:/opt/local/Library/Frameworks/Python.framework/Versions/2.7/bin:/Applications/ccp4-7.0/lib:/Applications/ccp4-7.0/bin:/Applications/ccp4-7.0:/usr/local/Cellar:/usr/local/bin:~/.emacs.d/bin:/Applications/emacs.app/Contents/MacOS:/usr/local/charmm/bin:/usr/local/go/bin:/Users/blaine/software/amber18/bin:/opt/X11/bin:/Applications/Julia-1.0.app/Contents/Resources/Julia/bin/julia:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/blaine/.gem/ruby/2.3.0/bin:/Applications/ATSAS/bin:/Applications/ATSAS/lib:/Applications/ATSAS/share:/Users/software/prody/scripts:/Users/blaine/software/XDS:/Users/blaine/software/xdsme/bin/noarch:/usr/local:/Users/blaine/.cask/bin:/Users/blaine/software/doconce/bin:/Users/blaine/software/rna-pdb-tools/bin:/Users/blaine/software/dock6/bin:/Users/blaine/software/apache-maven-3.5.2/bin:/usr/local/bin/R:/Users/blaine/software/x3dna-v2.3/bin:/Users/blaine/software/curves+:/Users/blaine/software/rna-pdb-tools:/Users/blaine/software/dakota-6.11.0/bin:/Users/blaine/software/dakota-6.11.0/share/dakota/test:/Users/blaine/software/dakota-6.11.0/gui/Contents/MacOS:/bin:/usr/local/Cellar/python/3.7.5/bin:/Users/blaine/.gem/ruby/2.6.0/bin/usr/local/gromacs/share/man:/Applications/ccp4-7.0/share/man:/usr/share/man:/usr/local/share/man:/opt/X11/share/man__RESULT"

To ensure normal operation, you should investigate and remove the
cause of the error in your initialization file.  Start Emacs with
the ‘--debug-init’ option to view a complete error backtrace.

**Comment by MooersLab** [comment]
Weird, the '--debug-init' flag is not working with v26.3 whereas it was earlier this week.

**Comment by freetonik** [comment]
Let's try something. Open `init.el` and find this section:

```
(use-package exec-path-from-shell
  :ensure t)
```

Replace it with this:

```
(use-package exec-path-from-shell
  :ensure t
  :config
  (setq explicit-shell-file-name "/usr/bin/env bash"))
```

and restart Emacs. See if the error disappears.

**Comment by MooersLab** [comment]
I get the same error message.
I launched emacs from a zsh shell.

**Comment by MooersLab** [comment]
If I switch to bash shell, I get the same warning in emacs.

**Comment by freetonik** [comment]
I think the problem originates with conda or some other Python package on your machine. [This](https://github.com/conda/conda/issues/8284) is probably related. 

The package `exec-path-from-shell` cannot handle the fact that the environment yields some unexpected output.

If you don't use `gfortran_osx-64`, you could try uninstalling or upgrading it.

**Comment by MooersLab** [comment]
Yes,  `deactivate-gfortran_osx-64.sh` is for a conda install of gfortran. I updated the gfortan with conda and the error message went away from both the zsh shell and emacs. 

In addition, I am now getting some of the automated formatting from castlemac of my org file
<img width="627" alt="castlemac" src="https://user-images.githubusercontent.com/15176203/87820565-e9d29200-c833-11ea-9e8d-76bbfecd2850.png">


However, C-c a is still not recognized.

As you may know, zsh is the default shell in Catalina.
